src/NLog/Internal/FileAppenders/BaseFileAppender.cs

```diff
@@ -177,12 +177,12 @@ protected FileStream CreateFileStream(bool allowConcurrentWrite)
                         }
 
                         Directory.CreateDirectory(Path.GetDirectoryName(this.FileName));
-                        return this.TryCreateFileStream(allowConcurrentWrite);
+                        return this.TryCreateFileStream(allowAtomicConcurrentWrite);
                     }
                 }
                 catch (IOException)
                 {
-                    if (!this.CreateFileParameters.ConcurrentWrites || !allowConcurrentWrite || i + 1 == this.CreateFileParameters.ConcurrentWriteAttempts)
```

Comment:
The `allowConcurrentWrite` parameter must apply only to how the stream is created when `TryCreateFileStream()` is called (that is, if true, then multiple processes may have an open stream for the file at the same time). 

The parameter must not affect the concurrent write retry. Write retry must still work when `allowConcurrentWrite = false`. Hence, the exception must not be thrown in this case.